### PR TITLE
Adds stampedes

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
:cl: Kor
add: When crossing over a downed person, you will deal brute damage to them.
add: When crossing over a downed person while running, there is a chance for you to trip.
/:cl:

I think this will cause panicked situations to get even more out of hand, and the accidental maiming and death that will result when people try to pile in through the broken escape shuttle window will be amusing and in the spirit of the game.

Hopline mobs fights will also be extra exciting when the clown gets stampeded to death while greyshirts are disarm spamming each other.

I should probably generalize this carbons, and I'm going to make it actually apply hits to areas instead of adjustbruteloss, but I wanted to see if this actually plays out well or not before putting more work in it.

### CODE REVIEW

**Comment by lzimann** [comment]
might want to add some admin logs to it

**Comment by ChangelingRain** [comment]
This should only happen on run intent; if you're walking, you can avoid stepping on people

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -215,6 +215,13 @@
 	if(istype(MB))
 		MB.RunOver(src)
 
+	if(ishuman(AM))
+		var/mob/living/carbon/human/stampede = AM
+		visible_message("<span class='danger'>[src] is trampled by [stampede.name]!</span>", "<span class='userdanger'>You got trampled by [stampede.name]!</span>")
+		adjustBruteLoss(8)
+		if(prob(30) && stampede.m_intent == MOVE_INTENT_RUN)
+			visible_message("<span class='danger'>[src] trips over [AM]!</span>")
```
Comment (swindly) [inline code/modules/mob/living/carbon/human/human.dm]: Is this backwards?

Comment (KorPhaeron) [inline code/modules/mob/living/carbon/human/human.dm]: Yes

**Comment by KorPhaeron** [comment]
I feel like if we add walking either

1) Everyone will just walk in a crowded situation, which will nullify the whole thing, even with 30 people shoving back and forth

2) Admins will just ban people for not walking in a crowded situation until number 1 happens

**Comment by KorPhaeron** [comment]
There should be logs though yeah

**Comment by kevinz000** [comment]
kor add tackling and dogpiles next lmao

**Comment by WJohn** [comment]
There will literally be two hundred logs generated in a matter of seconds whenever the shuttle arrives.

**Comment by ChangelingRain** [comment]
Also if you don't have legs you shouldn't trample people because you're just crawling

**Comment by KorPhaeron** [comment]
Good point

**Comment by Xhuis** [comment]
something something black friday

**Comment by anconfuzedrock** [comment]
good shit but seriously add the walking or SOMETHING

**Comment by dannno** [comment]
walking not doing it seems like a good idea imo

>kor add tackling and dogpiles next lmao

this

**Comment by sawrge** [comment]
>be chef
>escape shuttle just docked
>lobster escaped from kitchen, running to escape shuttle
>im wearing my cleats
>trampled over a bunch of assistants, killing them
>smash lobster into paste
>all my friends start calling me chef boy-r-dumb

**Comment by sawrge** [comment]
also add cleats that do bonus trample damage

**Comment by dannno** [comment]
DONT

**Comment by swindly** [comment]
New meta: rest when chased by someone with dual esword.

**Comment by sawrge** [comment]
nerw idea: replace dual esword with e-cleats which are normal shoes til you use them and they extend into eswords on your feat that deal 10x trample damage

**Comment by dannno** [comment]
on it

**Comment by Shapsy** [comment]
I feel like 8 brute damage per step is a really, really high number for how easy/quick it is to run over people

**Comment by dannno** [comment]
this ^

**Comment by kevinz000** [comment]
KORPHAERON PLEASE FOR THE LOVE OF GOD TEST YOUR CODE DONT BE A GOOFBALL WITH SERVER KEY THAT IS HORRIFYING!!

**Comment by dannno** [comment]
lol if you fucking swap people on help intent you trample them

**Comment by jughu** [comment]
Who needs punching disarm and stampede will be waaay stronger

**Comment by bgobandit** [comment]
>Everyone will just walk in a crowded situation, which will nullify the whole thing, even with 30 people shoving back and forth

the existence of people slipping over water, still, disproves this

**Comment by kevinz000** [comment]
@KorPhaeron Don't forget to make FLYING movement_type mobs not do this, because trampling everyone with a flightsuit is hilarious as fuck and overpowered.

**Comment by Grazyn** [comment]
Corpses should count for the trip chance too, gives geneticists a reason to keep the room clean and makes medbay hilarious during emergencies when everyone is tripping on the bodies piled everywhere

**Comment by kevinz000** [comment]
each body on a tile should increase trip  chance lmao

**Comment by Arianya** [comment]
Someone test how quickly you can kill someone while on stims + meth with this.

**Comment by Grazyn** [comment]
I think being pushed down in the middle of a full HoP line where everyone is on help intent (constantly swapping tile with each other) will kill you pretty quick.

On another note, martial arts should probably make you immune to tripping. It doesn't make sense for a kung fu master to casually send his opponent sprawling on the floor with an elegant move only to trip over him the next second like a dumbass.

**Comment by vcordie** [comment]
Shouldnt trampel on help intent, should instead cause you to move lsower through that tile.

**Comment by jughu** [comment]
The problem is that 8 brute will litteraly be better then a punch so you rather slip them or push them so you can trample their face in with speed adders like meth and such and basically have a faster damage then punching

File: code/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -215,6 +215,13 @@
 	if(istype(MB))
 		MB.RunOver(src)
 
+	if(ishuman(AM) && lying)
+		var/mob/living/carbon/human/stampede = AM
+		visible_message("<span class='danger'>[src] is trampled by [stampede.name]!</span>", "<span class='userdanger'>You got trampled by [stampede.name]!</span>")
+		adjustBruteLoss(8)
```
Comment (Incoming5643) [inline code/modules/mob/living/carbon/human/human.dm]: Doing it this way ignores armoring I think. Dark memories of mulebots stacking up 15 damage armor ignoring permastuns.

**Comment by KorPhaeron** [comment]
I'm not sure this is actually workable because

1) The trip mechanic gets very annoying. People have been running over bodies for years, so you tend to not worry about or avoid them

2) The damage mechanic would be horribly overpowered without the trip mechanic I think

Soliciting ideas to change this but the PR is probably going to be dead

File: code/modules/mob/living/carbon/human/human.dm (hunk 2 above)
Comment (KorPhaeron) [inline code/modules/mob/living/carbon/human/human.dm]: Yeah, was going to run armour and hit zones and stuff but didnt want to do that extra work if I couldn't get the rest of the mechanics to function

**Comment by ChangelingRain** [comment]
Let you walk to avoid doing damage!!
Make the damage, like, 3 and affected by armor, also.

**Comment by dannno** [comment]
If you do try to make this work, make the damage localized to a random body part as opposed to being like 1 brute spread across all body parts, because people eat up the brute shit because they're lazy

**Comment by KorPhaeron** [comment]
@dannno 

I was planning on it, I was just too lazy to do the targeting and armor checks since I had a feeling the whole thing might get scrapped anyway

**Comment by darkpaladin109** [comment]
Aww, I don't want this to be another cool little feature that just doesn't get merged.
Maybe speed boosting things like Meth could up your chance to trip up by a good bit? You're not gonna be too careful running over someone really fast.

**Comment by ShadowDeath6** [comment]
having a speedmod that's faster than normal should send you flying space lube style when you trip

**Comment by PKPenguin321** [comment]
Instead of slipping just make you not trample if you're on click cool down and have it apply click cool down so you can't mix it with regular punches

**Comment by PKPenguin321** [comment]
Also either make it so it doesn't happen on help intent and/or walking toggle

**Comment by KorPhaeron** [comment]
>click cooldown

oh duh that makes sense

**Comment by KorPhaeron** [comment]
I wish the tripping would work though, I want things to domino out of control with piles of people tumbling over each other

**Comment by Hatfish** [comment]
This is going to make lavaland hell with all the legion corpses in those tight 1 tile wide shafts.

**Comment by KorPhaeron** [comment]
Maybe only living people should trip

**Comment by Core0verload** [comment]
kor is just goof v2

File: code/modules/mob/living/carbon/human/human.dm [hunk 3]
```diff
@@ -215,6 +215,13 @@
 	if(istype(MB))
 		MB.RunOver(src)
 
+	if(ishuman(AM) && lying)
```
Comment (GunHog) [inline code/modules/mob/living/carbon/human/human.dm]: This should not apply if the player is:
- walking.
- is missing legs.
- flying or floating.
- in zero gravity.
- a dwarf.
- being thrown.
- on a vehicle (in which case MORE damage should occur!).

**Comment by Fox-McCloud** [comment]
How to instakill someone with this; push them over and spam open/close a locker with a macro while standing over them.

**Comment by Kromgar** [comment]
People use macros?

**Comment by PKPenguin321** [comment]
That won't work if it uses click cooldown

**Comment by Saegrimr** [comment]
RIP escape pod snuggle party, whoever lays down first is getting fucking destroyed.

File: code/modules/mob/living/carbon/human/human.dm (hunk 3 above)
Comment (ShadowDeath6) [inline code/modules/mob/living/carbon/human/human.dm]: >buffing manlet power even more

Comment (GunHog) [inline code/modules/mob/living/carbon/human/human.dm]: How is it a buff?

Comment (ShadowDeath6) [inline code/modules/mob/living/carbon/human/human.dm]: Unless you WANT to kill someone by spam stepping on them, it means you don't kill people by stepping on them

Comment (GunHog) [inline code/modules/mob/living/carbon/human/human.dm]: But I am asking that Dwarves are entirely UNABLE to trample someone, as they are too small. I ask again; How is this "buffing manlet power even more"?

**Comment by KorPhaeron** [comment]
Joan said she had ideas for this

**Pull request closed without merging**